.teamcity/subprojects.json

```diff
@@ -986,6 +986,13 @@
     "functionalTests": true,
     "crossVersionTests": false
   },
+  {
+    "name": "problems-rendering",
+    "path": "platforms/ide/problems-rendering",
```

Comment:
I am strongly against putting rendering functions in `problems-api`. It is an implementation detail that should be strictly separated from the API dependency.

On the other hand, the functionality cannot be put into `problems` either, as it would cause a circular dependency formed by `core`.

Reply:
Ultimately it's your code; I won't block the PR on this.